File: android/app/src/main/java/com/fuggel/Uway/model/Instruction.kt
package com.fuggel.Uway.model

data class Instruction(
    val maneuverLocation: List<Double>,
    val maneuverType: ManeuverType,
    val voiceInstructions: List<VoiceInstruction>,
)

data class VoiceInstruction(
    val distanceAlongGeometry: Double,
    val announcement: String,
    val ssmlAnnouncement: String,
    val type: String
)


enum class ManeuverType(val value: String) {
    TURN("turn"),
    DEPART("depart"),
    ARRIVE("arrive"),
    MERGE("merge"),
    ON_RAMP("on ramp"),
    OFF_RAMP("off ramp"),
    FORK("fork"),
    END_OF_ROAD("end of road"),
    CONTINUE("continue"),
    ROUNDABOUT("roundabout");

    companion object {
        fun fromValue(value: String): ManeuverType? {
            return entries.find { it.value == value }
        }
    }
}

File: ios/Models/Instruction.swift
//
//  Instruction.swift
//  Uway
//
//  Created by Furkan Ceylan on 20.06.25.
//

import Foundation

enum ManeuverType: String {
    case turn
    case depart
    case arrive
    case merge
    case onRamp = "on ramp"
    case offRamp = "off ramp"
    case fork
    case endOfRoad = "end of road"
    case `continue`
    case roundabout

    static func fromValue(_ value: String) -> ManeuverType {
        return ManeuverType(rawValue: value) ?? .continue
    }
}

struct VoiceInstruction {
    let distanceAlongGeometry: Double
    let announcement: String
    let ssmlAnnouncement: String
    let type: String
}

struct Instruction {
    let maneuverLocation: [Double]
    let maneuverType: ManeuverType
    let voiceInstructions: [VoiceInstruction]
}
